### PR TITLE
Add OrgGroups and roles

### PR DESCRIPTION
**Pull request by jansenk**
https://openedx.atlassian.net/browse/EDUCATOR-4151

Add OrgGroups
Add Organization permissions and roles
Tests

### CODE REVIEW

**Review by zacharis278 — APPROVED** [review]
lgtm

File: registrar/apps/api/v1/tests/test_views.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
     StaffUserFactory,
 )
 from registrar.apps.enrollments.models import Organization
-from registrar.apps.enrollments.tests.factories import (
+from registrar.apps.enrollments.tests.factories import (  # pylint: disable=no-name-in-module
```
Comment (kdmccormick) [inline registrar/apps/api/v1/tests/test_views.py]: pylint passes for me without these `no-name-in-module` suppressions. Was pylint failing locally for you, or on jenkins, or both?

Comment (jansenk) [reply registrar/apps/api/v1/tests/test_views.py]: It was failing for me locally

File: registrar/apps/enrollments/models.py [hunk 2]
```diff
@@ -45,6 +55,41 @@ def __str__(self):
...
+        app_label = 'enrollments'
+        verbose_name = 'Organization Group'
+
+    ROLE_CHOICES = (
+        (perms.OrganizationReadMetadataRole.name, 'Read an Organization Metadata'),
+        (perms.OrganizationReadEnrollmentsRole.name, 'Read Enrollments Data'),
+        (perms.OrganizationReadWriteEnrollmentsRole.name, 'Read and Write Enrollments Data'),
+    )
+
+    organization = models.ForeignKey(Organization, on_delete=models.CASCADE)
```
Comment (kdmccormick) [inline registrar/apps/enrollments/models.py:74]: If we're doing `organization = None` for global permissions, we need `null=True` here, along with a comment saying so.

Additionally, `OrgGroup.save` will need to be tweaked, and we should have a test verifying that `organization = None` implies global permissions.

File: registrar/apps/api/v1/tests/test_views.py (hunk 1 above)
Comment (kdmccormick) [inline registrar/apps/api/v1/tests/test_views.py:14]: pylint passes for me without these `no-name-in-module` suppressions. Was pylint failing locally for you, or on jenkins, or both?

**Review by kdmccormick — CHANGES_REQUESTED** [review]
Looks good other than the comments I left.

File: registrar/apps/enrollments/models.py [hunk 3]
```diff
@@ -45,6 +55,41 @@ def __str__(self):
         return self.name
 
 
+class OrgGroup(Group):
```
Comment (kdmccormick) [inline registrar/apps/enrollments/models.py]: If there's a way to remove the `Permissions` dialog from the admin panel, that'd be a good thing to do.

File: registrar/apps/enrollments/models.py [hunk 4]
```diff
@@ -45,6 +55,41 @@ def __str__(self):
...
+    Group subclass to grant select guardian permissions to a group on an organization level
+
+    .. no_pii::
+    """
+    class Meta(object):
+        app_label = 'enrollments'
+        verbose_name = 'Organization Group'
+
+    ROLE_CHOICES = (
+        (perms.OrganizationReadMetadataRole.name, 'Read an Organization Metadata'),
```
Comment (kdmccormick) [inline registrar/apps/enrollments/models.py]: I recommend changing this to `'Read Metadata Only'`, because:
* It's program metadata they're reading, not organization metadata
* Having `'Only'` in the name hints that the other two roles allow metadata reading too

File: registrar/apps/enrollments/models.py (hunk 2 above)
Comment (schenedx) [reply registrar/apps/enrollments/models.py:74]: For the statement:
`If we're doing organization = None for global permissions, we need null=True here, along with a comment saying so.`, I thought we just won't have a data row into this table if the group created is not limited to any organization, right?

Comment (iloveagent57) [reply registrar/apps/enrollments/models.py:74]: That's right, Simon - we'll use vanilla `auth_group` records for global perms - every `OrgGroup` should have a non-null organization.

File: registrar/apps/enrollments/models.py (hunk 3 above)
Comment (iloveagent57) [inline registrar/apps/enrollments/models.py]: Let the great debate of our time begin: `OrgGroup`, `OrganizationGroup`, or something else?  `AccessControlGroup`?  Also, does this belong in the `enrollments` app or the `core` app?  I think I can make a pretty good case that it belongs in `core`.

File: registrar/apps/enrollments/tests/test_models.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        super(OrgGroupTests, self).setUp()
+        self.organization = OrganizationFactory()
+        self.user = UserFactory()
+
+    @ddt.data(
+        perm.OrganizationReadMetadataRole,
+        perm.OrganizationReadEnrollmentsRole,
+        perm.OrganizationReadWriteEnrollmentsRole,
+    )
+    def test_roles(self, role):
```
Comment (schenedx) [inline registrar/apps/enrollments/tests/test_models.py:28]: Can we have another test to specifically make sure, if a user is created and saved into org 1, it do not have permissions for any permissions on org 2?

**Review by schenedx — COMMENTED** [review]
suggested one more test case.

File: registrar/apps/enrollments/models.py (hunk 3 above)
Comment (schenedx) [reply registrar/apps/enrollments/models.py]: I suggest `core` as well. I can see we have different django app for different data set to proxy with

File: registrar/apps/api/v1/tests/test_views.py (hunk 1 above)
Comment (jansenk) [reply registrar/apps/api/v1/tests/test_views.py:14]: Yeah, this was failing for me locally

File: registrar/apps/enrollments/models.py (hunk 3 above)
Comment (kdmccormick) [reply registrar/apps/enrollments/models.py]: My vote is `OrganizationGroup`. Unnecessary abbreviation is confusing.

I think it belongs in `core`, but then `Organization` should also be moved to `core`. `core` shouldn't be importing from `enrollments`.

File: registrar/apps/enrollments/models.py (hunk 2 above)
Comment (kdmccormick) [reply registrar/apps/enrollments/models.py:74]: Sounds good; my misunderstanding.

File: registrar/apps/enrollments/models.py (hunk 3 above)
Comment (schenedx) [reply registrar/apps/enrollments/models.py]: @kdmccormick I agree we should move Organization out of enrollment app. 
Can we just create a new ticket to do the move instead of adding scope to this PR? @jansenk do you mind creating the ticket to capture the conversation above?

Comment (schenedx) [reply registrar/apps/enrollments/models.py]: @jansenk I already created the move ticket at https://openedx.atlassian.net/browse/EDUCATOR-4184

**Review by schenedx — APPROVED** [review]
👍
